db/version_set.cc

```diff
@@ -1758,6 +1762,35 @@ void VersionStorageInfo::ComputeExpiredTtlFiles(
   }
 }
 
+void VersionStorageInfo::ComputeFilesMarkedForPeriodicCompaction(
+    const ImmutableCFOptions& ioptions,
+    const uint64_t periodic_compaction_time) {
+  assert(periodic_compaction_time > 0);
+
+  files_marked_for_periodic_compaction_.clear();
+
+  int64_t _current_time;
+  auto status = ioptions.env->GetCurrentTime(&_current_time);
+  if (!status.ok()) {
+    return;
+  }
+  const uint64_t current_time = static_cast<uint64_t>(_current_time);
+
+  for (int level = 0; level < num_levels(); level++) {
+    for (auto f : files_[level]) {
+      if (!f->being_compacted && f->fd.table_reader != nullptr &&
+          f->fd.table_reader->GetTableProperties() != nullptr) {
+        auto file_creation_time =
+            f->fd.table_reader->GetTableProperties()->file_creation_time;
+        if (file_creation_time > 0 &&
+            file_creation_time < (current_time - periodic_compaction_time)) {
```

Comment:
nit: `current_time - periodic_compaction_time` can be calculated before the for-loop.